repository/src/main/resources/alfresco/node-services-context.xml

```diff
@@ -330,6 +330,39 @@
         <property name="nodeService" ref="nodeService"/>
         <property name="policyComponent" ref="policyComponent"/>
         <property name="searchTrackingComponent" ref="searchTrackingComponent" />
-    </bean>   
+    </bean>
+
+    <bean id="nodeSizeThreadPool" class="org.alfresco.util.ThreadPoolExecutorFactoryBean">
+        <property name="poolName">
+            <value>defaultThreadPool</value>
+        </property>
+        <property name="corePoolSize">
+            <value>5</value>
+        </property>
+        <property name="maximumPoolSize">
+            <value>10</value>
+        </property>

```

Comment:
The property values should be externalised in `repository.properties` so that they can be overridden with appropriate values. As previously discussed, you must also set the `workQueueSize` property to a sensible number, such as 100, and this should be externalised as well. Leaving the default value of -1 (an unbounded queue) in an API is not advisable.

Reply:
As suggested,
Configured.